File: search-application/src/main/java/com/example/search/application/command/RecordSearchClickCommand.java
package com.example.search.application.command;

import com.example.search.domain.product.ProductId;

import java.util.Objects;

/**
 * 검색 결과 클릭 기록 명령. boost rule 학습을 위한 시그널.
 */
public record RecordSearchClickCommand(
        String searchId,
        String userId,
        ProductId productId,
        String keyword,
        int rank
) {

    public RecordSearchClickCommand {
        Objects.requireNonNull(searchId, "searchId");
        Objects.requireNonNull(productId, "productId");
        Objects.requireNonNull(keyword, "keyword");
        if (rank < 1) {
            throw new IllegalArgumentException("rank 1 이상: " + rank);
        }
    }
}

File: search-application/src/main/kotlin/com/example/search/application/command/RecordSearchClickCommand.kt
package com.example.search.application.command

import com.example.search.domain.product.ProductId

/**
 * 검색 결과 클릭 기록 명령. boost rule 학습을 위한 시그널.
 */
@JvmRecord
data class RecordSearchClickCommand(
    val searchId: String,
    val userId: String?,
    val productId: ProductId,
    val keyword: String,
    val rank: Int
) {
    init {
        if (rank < 1) {
            throw IllegalArgumentException("rank 1 이상: $rank")
        }
    }
}
